### PR TITLE
discovery: parallelize refresh

### PR DESCRIPTION
**Pull request by aleitner**
What: parallelize discovery refresh

Why: Discovery refresh checks nodes in a sequential for loop. Run the checks in parallel

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

**Comment by aleitner** [comment]
> Where are the tests?

https://github.com/storj/storj/blob/alex/node9/pkg/discovery/service_test.go

File: pkg/discovery/service.go [hunk 1]
```diff
@@ -54,8 +54,8 @@ func New(logger *zap.Logger, ol *overlay.Cache, kad *kademlia.Kademlia, config C
 		cache: ol,
 		kad:   kad,
 
-		refreshOffset: 0,
-		refreshLimit:  config.RefreshLimit,
+		refreshLimit: config.RefreshLimit,
+		refreshCount: config.RefreshCount,
```
Comment (egonelbre) [inline pkg/discovery/service.go]: refreshConcurrency

File: pkg/discovery/service.go [hunk 2]
```diff
@@ -97,54 +97,67 @@ func (discovery *Discovery) Run(ctx context.Context) (err error) {
...
+			node := node
+
+			limiter.Go(ctx, func() {
+				info, err := discovery.kad.FetchInfo(ctx, *node)
+				if ctx.Err() != nil {
+					return
+				}
+
+				if err != nil {
+					discovery.log.Info("could not ping node", zap.Stringer("ID", node.Id), zap.Error(err))
```
Comment (egonelbre) [inline pkg/discovery/service.go]: fetch info should be already calling update uptime internally

File: pkg/discovery/service.go [hunk 3]
```diff
@@ -97,54 +97,67 @@ func (discovery *Discovery) Run(ctx context.Context) (err error) {
 func (discovery *Discovery) refresh(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	list, more, err := discovery.cache.Paginate(ctx, discovery.refreshOffset, discovery.refreshLimit)
+	q, err := discovery.cache.CountQualified(ctx)
```
Comment (egonelbre) [inline pkg/discovery/service.go]: You shouldn't need the count.

File: storage/common.go [hunk 4]
```diff
@@ -45,7 +45,7 @@ type Values []Value
 type Items []ListItem
 
 // LookupLimit is enforced by storage implementations
-const LookupLimit = 1000
+const LookupLimit = 2000
```
Comment (egonelbre) [inline storage/common.go]: This lookup limit is used for multiple things, do not change this.

**Pull request merged**